### PR TITLE
Add TLS certificate assets

### PR DESCRIPTION
**Pull request by timopollmeier**

### CODE REVIEW

File: src/gmp_tls_certificates.c [hunk 1]
```diff
@@ -205,3 +205,223 @@ get_tls_certificates_run (gmp_parser_t *gmp_parser, GError **error)
...
+    }
+
+  /* Check given info. */
+
+  name = entity_child (entity, "name");
+  comment = entity_child (entity, "comment");
+  certificate = entity_child (entity, "certificate");
+
+  switch (create_tls_certificate
+                (name ? entity_text (comment) : NULL,
```
Comment (mattmundell) [inline src/gmp_tls_certificates.c]: should be entity_text (name), yes?

File: src/manage_sql_tls_certificates.c [hunk 2]
```diff
@@ -447,6 +447,256 @@ copy_tls_certificate (const char *name,
...
+  sql ("DELETE FROM permissions"
+       " WHERE resource_type = 'tls_certificate'"
+       " AND resource_location = %i"
+       " AND resource IN (SELECT id FROM tls_certificates_trash"
+       "                  WHERE owner = (SELECT id FROM users"
+       "                                 WHERE uuid = '%s'));",
+       LOCATION_TRASH,
+       current_credentials.uuid);
+
+  sql ("DELETE FROM tickets_trash"
```
Comment (mattmundell) [inline src/manage_sql_tls_certificates.c]: Should be tls_certificates_trash.

File: src/manage_sql_tls_certificates.c [hunk 3]
```diff
@@ -447,6 +447,256 @@ copy_tls_certificate (const char *name,
...
+       " WHERE owner = (SELECT id FROM users WHERE uuid = '%s');",
+       current_credentials.uuid);
+}
+
+/**
+ * @brief Delete all TLS certificate owner by a user.
```
Comment (mattmundell) [inline src/manage_sql_tls_certificates.c]: "owned"

File: src/manage_sql_tls_certificates.c [hunk 4]
```diff
@@ -447,6 +447,256 @@ copy_tls_certificate (const char *name,
...
+ */
+void
+inherit_tls_certificates (user_t user, user_t inheritor)
+{
+  /* Regular tls_certificate. */
+
+  sql ("UPDATE tls_certificate SET owner = %llu WHERE owner = %llu;",
+       inheritor, user);
+
+  /* Trash tickets. */
```
Comment (mattmundell) [inline src/manage_sql_tls_certificates.c]: "TLS certificates"

File: src/gmp_tls_certificates.c (hunk 1 above)
Comment (timopollmeier) [reply src/gmp_tls_certificates.c]: Correct, I've changed it in 106673d51b67f0c62e005f3a64236f27b81d4586.

File: src/manage_sql_tls_certificates.c (hunk 3 above)
Comment (timopollmeier) [reply src/manage_sql_tls_certificates.c]: Fixed in 106673d51b67f0c62e005f3a64236f27b81d4586

File: src/manage_sql_tls_certificates.c (hunk 2 above)
Comment (timopollmeier) [reply src/manage_sql_tls_certificates.c]: Fixed in 106673d51b67f0c62e005f3a64236f27b81d4586.

File: src/manage_sql_tls_certificates.c (hunk 4 above)
Comment (timopollmeier) [reply src/manage_sql_tls_certificates.c]: Fixed in 106673d51b67f0c62e005f3a64236f27b81d4586.

**Pull request merged**